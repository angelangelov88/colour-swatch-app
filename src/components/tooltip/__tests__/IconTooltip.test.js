import renderer from 'react-test-renderer'
import {IconTooltip} from "../IconTooltip"
import { BanIcon, BellIcon, DocumentIcon, InboxIcon, InformationCircleIcon } from '@heroicons/react/outline'

it('should render with info icon, text and pointerPosition default/bottom', () => {
  const component = renderer.create(
  <IconTooltip 
    Icon={InformationCircleIcon}
    children="Tooltip Text"
  />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});

it('should render with info icon, pointerPosition top of the icon and the text in children', () => {
  const component = renderer.create(
  <IconTooltip 
    Icon={InformationCircleIcon}
    pointerPosition="top"
    children="Tooltip Text"
  />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('should render with info icon and long text', () => {
  const component = renderer.create(
  <IconTooltip 
    Icon={InformationCircleIcon}
    children='Please use this field to add additional information Please use this field to add additional information Please use this field to add additional information Please use this field to add additional information Please use this field to add additional information Please use this field to add additional information Please use this field to add additional information Please use this field to add additional information Please use this field to add additional information Please use this field to add additional information Please use this field to add additional information Please use this field to add additional information '
  />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})


it('should render on click and with info icon', () => {
  const component = renderer.create(
  <IconTooltip 
    selected="click"
    Icon={InformationCircleIcon}
    children="Tooltip text"
  />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})


it('should render with hover, info icon and text on the right of the tooltip(left pointerPosition)', () => {
  const component = renderer.create(
    <IconTooltip 
      selected="hover"
      Icon={InformationCircleIcon}
      pointerPosition="left"
    >
      <h3 className="m-1 font-semibold">Info Header</h3>
      <p className="m-1 font-redhat">Please use this field to add additional information Please use this field to add additional information </p>
      <a href="https://test.com" className="m-1 text-blue-60 dark:text-blue-50">Link</a>
    </IconTooltip>
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})


it('should render with hover, Inbox icon, text and position bottom of the tooltip', () => {
  const component = renderer.create(
  <IconTooltip 
    selected="hover"
    Icon={InboxIcon}
    children="Tooltip Text"
    pointerPosition="top"
  />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})


it('should render with hover, document icon, text as html and pointerPosition bottom', () => {
  const component = renderer.create(
  <IconTooltip 
    selected="hover"
    Icon={DocumentIcon}
    pointerPosition="bottom"
  >
    <h1>Title</h1>
    <p>This is a paragraph</p>
    <a href="#">Link to nowhere</a>
  </IconTooltip>
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})


it('should render with hover, document icon, text as html and pointerPosition bottom with arrow position right', () => {
  const component = renderer.create(
  <IconTooltip 
    selected="hover"
    Icon={DocumentIcon}
    pointerPosition="bottom"
    arrowPosition="right"
  >
    <h1>Title</h1>
    <p>This is a paragraph</p>
    <a href="#">Link to nowhere</a>
  </IconTooltip>
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})


it('should render with click, inbox icon and html on the left', () => {
  const component = renderer.create(
  <IconTooltip 
    selected="click"
    Icon={InboxIcon}
    pointerPosition="right"
  >
    <h1>Title</h1>
    <p>This is the content</p>
    <p>This is the content</p>
    <p>This is the content</p>
    <a href="#">Link supposed to be here</a>
  </IconTooltip>
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})


it('should render with hover(default), info icon and text on the top', () => {
  const component = renderer.create(
  <IconTooltip 
    Icon={InformationCircleIcon}
    children="Tooltip Text"
    pointerPosition="bottom"
  />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})


it('should render with hover, document icon and text on top with right arrow', () => {
  const component = renderer.create(
  <IconTooltip 
    selected="hover"
    Icon={DocumentIcon}
    children="Tooltip Text"
    pointerPosition="bottom"
    arrowPosition="right"
  />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})


it('should render with hover, bell icon and text on top with left arrow', () => {
  const component = renderer.create(
  <IconTooltip 
    selected="hover"
    Icon={BellIcon}
    children="Tooltip Text"
    pointerPosition="bottom"
    arrowPosition="left"
  />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})


it('should render with click, bell icon and text with html on top with left arrow', () => {
  const component = renderer.create(
  <IconTooltip 
    selected="click"
    Icon={BellIcon}
    pointerPosition="bottom"
    arrowPosition="left"
  >
    <h3 className="m-1 font-semibold">Info Header</h3>
    <p className="m-1 font-redhat">Please use this field to add additional information Please use this field to add additional information </p>
    <a href="https://test.com" className="m-1 text-blue-60 dark:text-blue-50">Link</a>
  </IconTooltip>
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})



