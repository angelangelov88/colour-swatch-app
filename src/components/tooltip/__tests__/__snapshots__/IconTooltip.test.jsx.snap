// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render on click and with info icon 1`] = `
<div
  className="relative p-2 flex font-ibm text-gray-50 group"
>
  <svg
    aria-hidden="true"
    className="w-5 cursor-pointer"
    fill="none"
    onClick={[Function]}
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
  <div
    className="hidden absolute flex-col justify-evenly items-center p-2 rounded-lg shadow-[2px_4px_17px_-5px_rgba(97,97,98)] dark:bg-gray-130 text-gray-80 dark:text-gray-10 text-sm text-center after:absolute after:content:none after:border-[10px] after:border-transparent w-64 bottom-10 -left-[108px] -ml-0.5 "
  >
    <div>
      Tooltip Text
    </div>
    <svg
      className="absolute -bottom-2 left-25"
      fill="white"
      height="9"
      viewBox="0 0 28 9"
      width="28"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.1716 7.17158L6.82843 2.82843C5.01742 1.01742 2.56116 0 0 0H28C25.4388 0 22.9826 1.01741 21.1716 2.82842L16.8284 7.17157C15.2663 8.73367 12.7337 8.73367 11.1716 7.17158Z"
        fill="white"
      />
    </svg>
  </div>
</div>
`;

exports[`should render with click, bell icon and text with html on top with left arrow 1`] = `
<div
  className="relative p-2 flex font-ibm text-gray-50 group"
>
  <svg
    aria-hidden="true"
    className="w-5 cursor-pointer"
    fill="none"
    onClick={[Function]}
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M15 17h5l-1.405-1.405A2.032 2.032 0 0118 14.158V11a6.002 6.002 0 00-4-5.659V5a2 2 0 10-4 0v.341C7.67 6.165 6 8.388 6 11v3.159c0 .538-.214 1.055-.595 1.436L4 17h5m6 0v1a3 3 0 11-6 0v-1m6 0H9"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
  <div
    className="hidden absolute flex-col justify-evenly items-center p-2 rounded-lg shadow-[2px_4px_17px_-5px_rgba(97,97,98)] dark:bg-gray-130 text-gray-80 dark:text-gray-10 text-sm text-center after:absolute after:content:none after:border-[10px] after:border-transparent w-64 bottom-10 -left-[10px] "
  >
    <div>
      <h3
        className="m-1 font-semibold"
      >
        Info Header
      </h3>
      <p
        className="m-1 font-redhat"
      >
        Please use this field to add additional information Please use this field to add additional information 
      </p>
      <a
        className="m-1 text-blue-60 dark:text-blue-50"
        href="https://test.com"
      >
        Link
      </a>
    </div>
    <svg
      className="absolute -bottom-2 left-3.5"
      fill="white"
      height="9"
      viewBox="0 0 28 9"
      width="28"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.1716 7.17158L6.82843 2.82843C5.01742 1.01742 2.56116 0 0 0H28C25.4388 0 22.9826 1.01741 21.1716 2.82842L16.8284 7.17157C15.2663 8.73367 12.7337 8.73367 11.1716 7.17158Z"
        fill="white"
      />
    </svg>
  </div>
</div>
`;

exports[`should render with click, inbox icon and html on the left 1`] = `
<div
  className="relative p-2 flex font-ibm text-gray-50 group"
>
  <svg
    aria-hidden="true"
    className="w-5 cursor-pointer"
    fill="none"
    onClick={[Function]}
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20 13V6a2 2 0 00-2-2H6a2 2 0 00-2 2v7m16 0v5a2 2 0 01-2 2H6a2 2 0 01-2-2v-5m16 0h-2.586a1 1 0 00-.707.293l-2.414 2.414a1 1 0 01-.707.293h-3.172a1 1 0 01-.707-.293l-2.414-2.414A1 1 0 006.586 13H4"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
  <div
    className="hidden absolute flex-col justify-evenly items-center p-2 rounded-lg shadow-[2px_4px_17px_-5px_rgba(97,97,98)] dark:bg-gray-130 text-gray-80 dark:text-gray-10 text-sm text-center after:absolute after:content:none after:border-[10px] after:border-transparent max-w-xs h-40 -top-[62px] -left-[322px] -mt-0.5 "
  >
    <div>
      <h1>
        Title
      </h1>
      <p>
        This is the content
      </p>
      <p>
        This is the content
      </p>
      <p>
        This is the content
      </p>
      <a
        href="#"
      >
        Link supposed to be here
      </a>
    </div>
    <svg
      className="absolute bottom-[63px] -right-[9px]"
      fill="none"
      height="29"
      viewBox="0 0 10 29"
      width="10"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8.00001 17.1716L3.65686 21.5147C1.84585 23.3257 0.82843 25.782 0.82843 28.3431L0.82843 0.34314C0.82843 2.9043 1.84584 5.36055 3.65685 7.17156L8 11.5147C9.5621 13.0768 9.5621 15.6095 8.00001 17.1716Z"
        fill="white"
      />
    </svg>
  </div>
</div>
`;

exports[`should render with hover(default), info icon and text on the top 1`] = `
<div
  className="relative p-2 flex font-ibm text-gray-50 group hover:text-gray-70"
>
  <svg
    aria-hidden="true"
    className="w-5 cursor-pointer"
    fill="none"
    onClick={[Function]}
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
  <div
    className="hidden group-hover:flex absolute flex-col justify-evenly items-center p-2 rounded-lg shadow-[2px_4px_17px_-5px_rgba(97,97,98)] dark:bg-gray-130 text-gray-80 dark:text-gray-10 text-sm text-center after:absolute after:content:none after:border-[10px] after:border-transparent w-64 bottom-10 -left-[108px] -ml-0.5 "
  >
    <div>
      Tooltip Text
    </div>
    <svg
      className="absolute -bottom-2 left-25"
      fill="white"
      height="9"
      viewBox="0 0 28 9"
      width="28"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.1716 7.17158L6.82843 2.82843C5.01742 1.01742 2.56116 0 0 0H28C25.4388 0 22.9826 1.01741 21.1716 2.82842L16.8284 7.17157C15.2663 8.73367 12.7337 8.73367 11.1716 7.17158Z"
        fill="white"
      />
    </svg>
  </div>
</div>
`;

exports[`should render with hover, Inbox icon, text and position bottom of the tooltip 1`] = `
<div
  className="relative p-2 flex font-ibm text-gray-50 group hover:text-gray-70"
>
  <svg
    aria-hidden="true"
    className="w-5 cursor-pointer"
    fill="none"
    onClick={[Function]}
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20 13V6a2 2 0 00-2-2H6a2 2 0 00-2 2v7m16 0v5a2 2 0 01-2 2H6a2 2 0 01-2-2v-5m16 0h-2.586a1 1 0 00-.707.293l-2.414 2.414a1 1 0 01-.707.293h-3.172a1 1 0 01-.707-.293l-2.414-2.414A1 1 0 006.586 13H4"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
  <div
    className="hidden group-hover:flex absolute flex-col justify-evenly items-center p-2 rounded-lg shadow-[2px_4px_17px_-5px_rgba(97,97,98)] dark:bg-gray-130 text-gray-80 dark:text-gray-10 text-sm text-center after:absolute after:content:none after:border-[10px] after:border-transparent w-64 top-10 -left-[105px] -ml-0.5 "
  >
    <div>
      Tooltip Text
    </div>
    <svg
      className="absolute -top-[8px] left-[111px]"
      fill="none"
      height="9"
      viewBox="0 0 29 9"
      width="29"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M17.3284 1.17156L21.6716 5.51471C23.4826 7.32572 25.9388 8.34314 28.5 8.34314L0.5 8.34314C3.06116 8.34314 5.51741 7.32573 7.32842 5.51472L11.6716 1.17157C13.2337 -0.390531 15.7663 -0.390533 17.3284 1.17156Z"
        fill="white"
      />
    </svg>
  </div>
</div>
`;

exports[`should render with hover, bell icon and text on top with left arrow 1`] = `
<div
  className="relative p-2 flex font-ibm text-gray-50 group hover:text-gray-70"
>
  <svg
    aria-hidden="true"
    className="w-5 cursor-pointer"
    fill="none"
    onClick={[Function]}
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M15 17h5l-1.405-1.405A2.032 2.032 0 0118 14.158V11a6.002 6.002 0 00-4-5.659V5a2 2 0 10-4 0v.341C7.67 6.165 6 8.388 6 11v3.159c0 .538-.214 1.055-.595 1.436L4 17h5m6 0v1a3 3 0 11-6 0v-1m6 0H9"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
  <div
    className="hidden group-hover:flex absolute flex-col justify-evenly items-center p-2 rounded-lg shadow-[2px_4px_17px_-5px_rgba(97,97,98)] dark:bg-gray-130 text-gray-80 dark:text-gray-10 text-sm text-center after:absolute after:content:none after:border-[10px] after:border-transparent w-64 bottom-10 -left-[10px] "
  >
    <div>
      Tooltip Text
    </div>
    <svg
      className="absolute -bottom-2 left-3.5"
      fill="white"
      height="9"
      viewBox="0 0 28 9"
      width="28"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.1716 7.17158L6.82843 2.82843C5.01742 1.01742 2.56116 0 0 0H28C25.4388 0 22.9826 1.01741 21.1716 2.82842L16.8284 7.17157C15.2663 8.73367 12.7337 8.73367 11.1716 7.17158Z"
        fill="white"
      />
    </svg>
  </div>
</div>
`;

exports[`should render with hover, document icon and text on top with right arrow 1`] = `
<div
  className="relative p-2 flex font-ibm text-gray-50 group hover:text-gray-70"
>
  <svg
    aria-hidden="true"
    className="w-5 cursor-pointer"
    fill="none"
    onClick={[Function]}
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M7 21h10a2 2 0 002-2V9.414a1 1 0 00-.293-.707l-5.414-5.414A1 1 0 0012.586 3H7a2 2 0 00-2 2v14a2 2 0 002 2z"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
  <div
    className="hidden group-hover:flex absolute flex-col justify-evenly items-center p-2 rounded-lg shadow-[2px_4px_17px_-5px_rgba(97,97,98)] dark:bg-gray-130 text-gray-80 dark:text-gray-10 text-sm text-center after:absolute after:content:none after:border-[10px] after:border-transparent w-64 bottom-10 -left-52 -ml-0.5 "
  >
    <div>
      Tooltip Text
    </div>
    <svg
      className="absolute -bottom-2 right-3.5"
      fill="white"
      height="9"
      viewBox="0 0 28 9"
      width="28"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.1716 7.17158L6.82843 2.82843C5.01742 1.01742 2.56116 0 0 0H28C25.4388 0 22.9826 1.01741 21.1716 2.82842L16.8284 7.17157C15.2663 8.73367 12.7337 8.73367 11.1716 7.17158Z"
        fill="white"
      />
    </svg>
  </div>
</div>
`;

exports[`should render with hover, document icon, text as html and pointerPosition bottom 1`] = `
<div
  className="relative p-2 flex font-ibm text-gray-50 group hover:text-gray-70"
>
  <svg
    aria-hidden="true"
    className="w-5 cursor-pointer"
    fill="none"
    onClick={[Function]}
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M7 21h10a2 2 0 002-2V9.414a1 1 0 00-.293-.707l-5.414-5.414A1 1 0 0012.586 3H7a2 2 0 00-2 2v14a2 2 0 002 2z"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
  <div
    className="hidden group-hover:flex absolute flex-col justify-evenly items-center p-2 rounded-lg shadow-[2px_4px_17px_-5px_rgba(97,97,98)] dark:bg-gray-130 text-gray-80 dark:text-gray-10 text-sm text-center after:absolute after:content:none after:border-[10px] after:border-transparent w-64 bottom-10 -left-[108px] -ml-0.5 "
  >
    <div>
      <h1>
        Title
      </h1>
      <p>
        This is a paragraph
      </p>
      <a
        href="#"
      >
        Link to nowhere
      </a>
    </div>
    <svg
      className="absolute -bottom-2 left-25"
      fill="white"
      height="9"
      viewBox="0 0 28 9"
      width="28"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.1716 7.17158L6.82843 2.82843C5.01742 1.01742 2.56116 0 0 0H28C25.4388 0 22.9826 1.01741 21.1716 2.82842L16.8284 7.17157C15.2663 8.73367 12.7337 8.73367 11.1716 7.17158Z"
        fill="white"
      />
    </svg>
  </div>
</div>
`;

exports[`should render with hover, document icon, text as html and pointerPosition bottom with arrow position right 1`] = `
<div
  className="relative p-2 flex font-ibm text-gray-50 group hover:text-gray-70"
>
  <svg
    aria-hidden="true"
    className="w-5 cursor-pointer"
    fill="none"
    onClick={[Function]}
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M7 21h10a2 2 0 002-2V9.414a1 1 0 00-.293-.707l-5.414-5.414A1 1 0 0012.586 3H7a2 2 0 00-2 2v14a2 2 0 002 2z"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
  <div
    className="hidden group-hover:flex absolute flex-col justify-evenly items-center p-2 rounded-lg shadow-[2px_4px_17px_-5px_rgba(97,97,98)] dark:bg-gray-130 text-gray-80 dark:text-gray-10 text-sm text-center after:absolute after:content:none after:border-[10px] after:border-transparent w-64 bottom-10 -left-52 -ml-0.5 "
  >
    <div>
      <h1>
        Title
      </h1>
      <p>
        This is a paragraph
      </p>
      <a
        href="#"
      >
        Link to nowhere
      </a>
    </div>
    <svg
      className="absolute -bottom-2 right-3.5"
      fill="white"
      height="9"
      viewBox="0 0 28 9"
      width="28"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.1716 7.17158L6.82843 2.82843C5.01742 1.01742 2.56116 0 0 0H28C25.4388 0 22.9826 1.01741 21.1716 2.82842L16.8284 7.17157C15.2663 8.73367 12.7337 8.73367 11.1716 7.17158Z"
        fill="white"
      />
    </svg>
  </div>
</div>
`;

exports[`should render with hover, info icon and text on the right of the tooltip(left pointerPosition) 1`] = `
<div
  className="relative p-2 flex font-ibm text-gray-50 group hover:text-gray-70"
>
  <svg
    aria-hidden="true"
    className="w-5 cursor-pointer"
    fill="none"
    onClick={[Function]}
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
  <div
    className="hidden group-hover:flex absolute flex-col justify-evenly items-center p-2 rounded-lg shadow-[2px_4px_17px_-5px_rgba(97,97,98)] dark:bg-gray-130 text-gray-80 dark:text-gray-10 text-sm text-center after:absolute after:content:none after:border-[10px] after:border-transparent max-w-xl h-40 -top-[64px] left-10 "
  >
    <div>
      <h3
        className="m-1 font-semibold"
      >
        Info Header
      </h3>
      <p
        className="m-1 font-redhat"
      >
        Please use this field to add additional information Please use this field to add additional information 
      </p>
      <a
        className="m-1 text-blue-60 dark:text-blue-50"
        href="https://test.com"
      >
        Link
      </a>
    </div>
    <svg
      className="absolute bottom-[63px] -left-[9px]"
      fill="none"
      height="29"
      viewBox="0 0 10 29"
      width="10"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1.99999 11.5147L6.34314 7.17157C8.15415 5.36056 9.17157 2.9043 9.17157 0.34314L9.17157 28.3431C9.17157 25.782 8.15416 23.3257 6.34315 21.5147L2 17.1716C0.4379 15.6095 0.437897 13.0768 1.99999 11.5147Z"
        fill="white"
      />
    </svg>
  </div>
</div>
`;

exports[`should render with info icon and long text 1`] = `
<div
  className="relative p-2 flex font-ibm text-gray-50 group hover:text-gray-70"
>
  <svg
    aria-hidden="true"
    className="w-5 cursor-pointer"
    fill="none"
    onClick={[Function]}
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
  <div
    className="hidden group-hover:flex absolute flex-col justify-evenly items-center p-2 rounded-lg shadow-[2px_4px_17px_-5px_rgba(97,97,98)] dark:bg-gray-130 text-gray-80 dark:text-gray-10 text-sm text-center after:absolute after:content:none after:border-[10px] after:border-transparent w-64 bottom-10 -left-[108px] -ml-0.5 "
  >
    <div>
      Please use this field to add additional information Please use this field to add additional information Please use this field to add additional information Please use this field to add additional information Please use this field to add additional information Please use this field to add additional information Please use this field to add additional information Please use this field to add additional information Please use this field to add additional information Please use this field to add additional information Please use this field to add additional information Please use this field to add additional information
    </div>
    <svg
      className="absolute -bottom-2 left-25"
      fill="white"
      height="9"
      viewBox="0 0 28 9"
      width="28"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.1716 7.17158L6.82843 2.82843C5.01742 1.01742 2.56116 0 0 0H28C25.4388 0 22.9826 1.01741 21.1716 2.82842L16.8284 7.17157C15.2663 8.73367 12.7337 8.73367 11.1716 7.17158Z"
        fill="white"
      />
    </svg>
  </div>
</div>
`;

exports[`should render with info icon, pointerPosition top of the icon and the text in children 1`] = `
<div
  className="relative p-2 flex font-ibm text-gray-50 group hover:text-gray-70"
>
  <svg
    aria-hidden="true"
    className="w-5 cursor-pointer"
    fill="none"
    onClick={[Function]}
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
  <div
    className="hidden group-hover:flex absolute flex-col justify-evenly items-center p-2 rounded-lg shadow-[2px_4px_17px_-5px_rgba(97,97,98)] dark:bg-gray-130 text-gray-80 dark:text-gray-10 text-sm text-center after:absolute after:content:none after:border-[10px] after:border-transparent w-64 top-10 -left-[105px] -ml-0.5 "
  >
    <div>
      Tooltip Text
    </div>
    <svg
      className="absolute -top-[8px] left-[111px]"
      fill="none"
      height="9"
      viewBox="0 0 29 9"
      width="29"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M17.3284 1.17156L21.6716 5.51471C23.4826 7.32572 25.9388 8.34314 28.5 8.34314L0.5 8.34314C3.06116 8.34314 5.51741 7.32573 7.32842 5.51472L11.6716 1.17157C13.2337 -0.390531 15.7663 -0.390533 17.3284 1.17156Z"
        fill="white"
      />
    </svg>
  </div>
</div>
`;

exports[`should render with info icon, text and pointerPosition default/bottom 1`] = `
<div
  className="relative p-2 flex font-ibm text-gray-50 group hover:text-gray-70"
>
  <svg
    aria-hidden="true"
    className="w-5 cursor-pointer"
    fill="none"
    onClick={[Function]}
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
  <div
    className="hidden group-hover:flex absolute flex-col justify-evenly items-center p-2 rounded-lg shadow-[2px_4px_17px_-5px_rgba(97,97,98)] dark:bg-gray-130 text-gray-80 dark:text-gray-10 text-sm text-center after:absolute after:content:none after:border-[10px] after:border-transparent w-64 bottom-10 -left-[108px] -ml-0.5 "
  >
    <div>
      Tooltip Text
    </div>
    <svg
      className="absolute -bottom-2 left-25"
      fill="white"
      height="9"
      viewBox="0 0 28 9"
      width="28"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.1716 7.17158L6.82843 2.82843C5.01742 1.01742 2.56116 0 0 0H28C25.4388 0 22.9826 1.01741 21.1716 2.82842L16.8284 7.17157C15.2663 8.73367 12.7337 8.73367 11.1716 7.17158Z"
        fill="white"
      />
    </svg>
  </div>
</div>
`;
